pkg/scheduler/nodeinfo/node_info.go

```diff
@@ -471,11 +482,24 @@ func (n *NodeInfo) Clone() *NodeInfo {
 // VolumeLimits returns volume limits associated with the node
 func (n *NodeInfo) VolumeLimits() map[v1.ResourceName]int64 {
 	volumeLimits := map[v1.ResourceName]int64{}
+
 	for k, v := range n.AllocatableResource().ScalarResources {
 		if v1helper.IsAttachableVolumeResourceName(k) {
 			volumeLimits[k] = v
 		}
 	}
+
+	if n.csiNode != nil {
+		for i := range n.csiNode.Spec.Drivers {
+			d := n.csiNode.Spec.Drivers[i]
+			if d.Allocatable != nil && d.Allocatable.Count != nil {
+				// TODO: drop GetCSIAttachLimitKey once we don't get values from Node object
+				k := v1.ResourceName(volumeutil.GetCSIAttachLimitKey(d.Name))
```

Comment:
Can we establish map from ResourceName to int64 once so that we don't perform the calculation here every time ?

Reply:
What if we may need the ResourceName other place?